### PR TITLE
Create resource type and related role/scope for spaces

### PR DESCRIPTION
**Pull request by sbose78**
fixes https://github.com/fabric8-services/fabric8-auth/issues/383

To start using the authorization service for the Collaborators API, this PR -
- adds a new `resource_type` for spaces 

For the above resource type : 
- adds a role named `contributor` with scopes `contribute` , `view`
- adds a role named `viewer` with scopes named `view`
- adds a role named `admin` with scopes named `contribute` , `admin` ,`view`

### CODE REVIEW

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-384-1`

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-384-2`

**Comment by sbose78** [comment]
[test]

File: migration/sql-files/023-add-space-resourcetype.sql [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+             NAME, 
+             created_at) 
+VALUES     ('6422fda4-a0fa-4d3c-8b79-8061e5c05e12', 
+            'openshift.io/resource/space', 
+            Now()); 
+
+
+-- create a role named 'collaborator'
+
+INSERT INTO role
```
Comment (sbryzak) [inline migration/sql-files/023-add-space-resourcetype.sql]: I am concerned about the auth team deciding which production-level roles and scopes we are adding to the database without first discussing with other stakeholders - perhaps we could hold off on this for now?

Comment (sbose78) [reply migration/sql-files/023-add-space-resourcetype.sql]: I understand the concern, @sbryzak . 
This is meant to replace the keycloak-authz being used in the existing Add/List Collaborator API.

Irrespective of what the discussion goes with PM, we are gonna have to retain the same behavior for a while - and these changes will help to do so.

**Comment by sbose78** [comment]
@alexeykazakov , sure - this PR is meant to be start the discussion around it :) 
Btw, we could add the space resource type, but without the basic role & scope which is needed for existing collaborators, the space resource type isn't of much use.

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-384-3`

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-384-4`

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-384-5`

**Comment by sbryzak** [comment]
@sbose78 Could you please update this PR with the scopes that are documented in the spreadsheet? I think it would then be safe to proceed with this. @alexeykazakov do you agree with this?

**Comment by alexeykazakov** [comment]
@sbryzak +1

And btw I've updated the spreadsheet. Added orgs/groups/role assignment etc. Please review @sbose78 @sbryzak

**Comment by sbryzak** [comment]
[test]

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-384-7`

**Pull request merged**